pkg/resource/training_job/hooks.go

```diff
@@ -64,4 +69,23 @@ func (rm *resourceManager) customSetOutput(r *resource) {
 	}
 
 	svccommon.SetSyncedCondition(r, trainingJobStatus, &resourceName, &trainingJobModifyingStatuses)
+
+	warmpoolUsed := ackcompare.IsNotNil(r.ko.Spec.ResourceConfig) && ackcompare.IsNotNil(r.ko.Spec.ResourceConfig.KeepAlivePeriodInSeconds)
+
+	// Only requeue when warm pool is being used and when training job is in the completed state.
+	// WP will always have terminated status on error(Training Job or Warmpool).
+	if ackcompare.IsNotNil(trainingJobStatus) && *trainingJobStatus == svcsdk.TrainingJobStatusCompleted &&
+		warmpoolUsed {
+
+		// Sometimes DescribeTrainingJob does not contain the warm pool status
+		// In this condition the only possible status is Available or Terminated.
```

Comment:
I see we are handling the case where only possible Status is Available, what about terminated? Is terminated a code path we do not expect to be possible to end up in this code logic?

Reply:
Handled in line 88